src/hotspot/share/gc/shenandoah/shenandoahLock.hpp

```diff
@@ -42,6 +42,17 @@ class ShenandoahLock  {
 
   template<bool ALLOW_BLOCK>
   void contended_lock_internal(JavaThread* java_thread);
+
+  void yield_or_sleep(int &yields) {
+    if (yields < 5) {
+      os::naked_yield();
```

Comment:
Need `#include "runtime/os.hpp"` for this. There is likely a transitive dependency now, but it is cleaner to depend explicitly. Or, maybe move this definition to `shenandoahLock.cpp`, it would be even cleaner then, I think.